### ISSUE TITLE
[REQUEST] Internal API to access drun desktop entry cache from other modes

### ISSUE THREAD

**Issue by alebastr** [davatorium/rofi#1935]
### Before opening a feature request

- [X] I checked the *next* branch to see if the feature has already been implemented
- [X] I searched existing reports to see if it is already requested.

### What is the user problem or growth opportunity you want to see solved?

In absence of `_NET_WM_ICON` or `WM_ICON_NAME`, the recommended path to the application icon for window mode is `WM_CLASS` (or Wayland app_id) -> $WM_CLASS.desktop -> Icon.

For an added fun, it could be necessary to match by `StartupWMClass` instead of the .desktop filename (e.g. `org.codeberg.dnkl.foot.desktop` with `StartupWMClass=foot` and app_id `foot`).

It wouldn't be too complicated to duplicate the lookup logic, but the code already exists, with all the optimizations and file-based cache. It just needs to be exposed to other modes.

### How do you know that this problem exists today? Why is this important?

`window` mode both here and in the Wayland fork assumes that the icon name is a lower-case of `WM_CLASS` or `app_id`. That's not always true and we we're not able to display an icon for affected applications.

### Who will benefit from it?

Mostly Wayland fork. But XCB window mode could leverage the same logic to resolve the icons from WM_CLASS.

### Rofi version (rofi -v)

1.7.5

### Configuration

-

**Comment by DaveDavenport** [comment]
I agree it would be nice, but not sure how this would be implemented in a way it does not break some of the abstractions that are in the modes/plugins. As mentioned in previous issue, I don't really want to break this.

I think if somebody is willing to do this, we should move the desktop file parser into a separate part with an api  that in the background get used by both drun and window..

**Comment by oredaze** [comment]
I was about to make a new request, but I guess this one is close enough.

It would be **really** useful if we have unified icons between drun and window in the same way that a dock, for example plank, would do. This way I am not confused when I see a different icon from what I launched.
This will **greatly** increase the intuitiveness of window modi.

Those docks don't use NET_WM_ICON, instead they somehow pull icon names from .desktop files even if software doesn't get launched from a .desktop file.
I don't know how they do it, maybe check plank for that.

Having the wrong icon is like typing on a new keyboard layout, it totally kills the flow.

**Comment by DaveDavenport** [comment]
Please follow the guidelines and DO NOT add not directly related information/new issue to an existing issue.
Please create a new issue.  I agree it would be nice, and if you have a way to implement that, I would like to do that.

This issue would then be step 2. 

(I will mark these two messages as 'offtopic' soon).

**Comment by DaveDavenport** [comment]
New issue created. #2125
